core/src/main/java/me/lokka30/treasury/plugin/core/hooks/placeholder/economy/BalanceCache.java

```diff
@@ -2,49 +2,53 @@
  * This file is/was part of Treasury. To read more information about Treasury such as its licensing, see <https://github.com/ArcanePlugins/Treasury>.
  */
 
-package me.lokka30.treasury.plugin.bukkit.hooks.papi.economy;
+package me.lokka30.treasury.plugin.core.hooks.placeholder.economy;
 
 import com.google.common.collect.HashMultimap;
 import com.google.common.collect.Multimap;
 import java.math.BigDecimal;
 import java.util.AbstractMap;
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.Collection;
 import java.util.List;
 import java.util.Map;
 import java.util.UUID;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicReference;
 import me.lokka30.treasury.api.common.misc.FutureHelper;
 import me.lokka30.treasury.api.common.misc.TriState;
 import me.lokka30.treasury.api.common.response.TreasuryException;
 import me.lokka30.treasury.api.economy.EconomyProvider;
 import me.lokka30.treasury.api.economy.account.AccountData;
 import me.lokka30.treasury.api.economy.currency.Currency;
-import me.lokka30.treasury.plugin.bukkit.TreasuryBukkit;
 import me.lokka30.treasury.plugin.core.TreasuryPlugin;
-import org.bukkit.Bukkit;
-import org.bukkit.OfflinePlayer;
-import org.bukkit.scheduler.BukkitRunnable;
+import me.lokka30.treasury.plugin.core.hooks.PlayerData;
+import me.lokka30.treasury.plugin.core.hooks.placeholder.PlaceholdersExpansion;
+import me.lokka30.treasury.plugin.core.schedule.Scheduler;
 import org.jetbrains.annotations.Nullable;
 
-public class BalanceCache extends BukkitRunnable {
+public class BalanceCache extends Scheduler.ScheduledTask {
 
+    private final PlaceholdersExpansion base;
     private final Multimap<UUID, Map.Entry<String, BigDecimal>> balances = HashMultimap.create();
     private final int delay;
     private final AtomicReference<EconomyProvider> providerRef;
     private final AtomicReference<CountDownLatch> doneLatch = new AtomicReference<>(new CountDownLatch(
             1));
 
-    public BalanceCache(int delay, AtomicReference<EconomyProvider> providerRef) {
+    public BalanceCache(
+            PlaceholdersExpansion base, int delay, AtomicReference<EconomyProvider> providerRef
```

Comment:
Unsure if this was automatically formatted, the parameters list here looks quite ugly 😁